File: algorithms/1897.redistribute-characters-to-make-all-strings-equal/solution.rs
impl Solution {
    pub fn make_equal(words: Vec<String>) -> bool {
        let mut count = vec![0; 26];
        let len = words.len();

        for word in words {
            for c in word.chars() {
                count[c as usize - 'a' as usize] += 1;
            }
        }

        for x in count {
            if x % len != 0 {
                return false;
            }
        }

        true
    }
}

File: algorithms/1897.redistribute-characters-to-make-all-strings-equal/solution.cpp
#include <string>
#include <vector>

using namespace std;

class Solution {
  public:
    bool makeEqual(vector<string>& words) {
        if (words.size() == 1) {
            return true;
        }

        vector<int> counter(26, 0);
        for (const string& word : words) {
            for (char c : word) {
                counter[c - 'a']++;
            }
        }

        for (int i : counter) {
            if (i % words.size() != 0) {
                return false;
            }
        }

        return true;
    }
};
